common/lib/capa/capa/capa_problem.py

```diff
@@ -942,11 +942,18 @@ def response_a11y_data(self, response, inputfields, responsetype_id, problem_dat
                 # store <label> tag containing question text to delete
                 # it later otherwise question will be rendered twice
                 element_to_be_deleted = responsetype_label_tag
-            elif 'label' not in inputfields[0].attrib:
+            elif 'label' in inputfields[0].attrib:
+                # Get first <p> tag before responsetype, this <p> may contains the question text.
+                p_tag = response.xpath('preceding-sibling::*[1][self::p]')
+
+                if p_tag and p_tag[0].text == inputfields[0].attrib['label']:
+                    label = p_tag[0].text
+                    element_to_be_deleted = p_tag[0]
+            else:
                 # In this case the problems don't have tag or label attribute inside the responsetype
                 # so we will get the first preceding label tag w.r.t to this responsetype.
                 # This will take care of those multi-question problems that are not using --- in their markdown.
-                label_tag = response.xpath('preceding-sibling::label[1]')
+                label_tag = response.xpath('preceding-sibling::*[1][self::label]')
```

Comment:
exactly first preceding label tag, if something exist in between then we won't pick up.